frontend/src/scenes/plugins/Plugins.tsx

```diff
@@ -6,72 +6,80 @@ import { InstalledTab } from 'scenes/plugins/tabs/installed/InstalledTab'
 import { useActions, useValues } from 'kea'
 import { userLogic } from 'scenes/userLogic'
 import { pluginsLogic } from './pluginsLogic'
-import { Spin, Tabs, Tag } from 'antd'
-import { OptInPlugins } from 'scenes/plugins/optin/OptInPlugins'
+import { Alert, Tabs } from 'antd'
 import { PageHeader } from 'lib/components/PageHeader'
 import { PluginTab } from 'scenes/plugins/types'
 import { AdvancedTab } from 'scenes/plugins/tabs/advanced/AdvancedTab'
 import { canGloballyManagePlugins, canInstallPlugins, canViewPlugins } from './access'
 
 export function Plugins(): JSX.Element | null {
     const { user } = useValues(userLogic)
+    const { userUpdateRequest } = useActions(userLogic)
     const { pluginTab } = useValues(pluginsLogic)
     const { setPluginTab } = useActions(pluginsLogic)
     const { TabPane } = Tabs
 
-    if (!user) {
-        return <Spin />
-    }
+    useEffect(() => {
+        if (user) {
+            if (!canViewPlugins(user.organization)) {
+                window.location.href = '/'
+                return
+            }
+            if (!user.flags.has_checked_out_plugins) {
```

Comment:
Code organization note/suggestion: Move this into an action within pluginsLogic, e.g. `saveHasViewedPlugins`. This keeps business logic out of the view layer.

Not a priority though.

Reply:
Not relevant as this has been removed.